ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt

```diff
@@ -0,0 +1,170 @@
+package com.braintreepayments.api
+
+import androidx.annotation.VisibleForTesting
+import com.braintreepayments.api.ShopperInsightsAnalytics.GET_RECOMMENDED_PAYMENTS_FAILED
+import com.braintreepayments.api.ShopperInsightsAnalytics.GET_RECOMMENDED_PAYMENTS_STARTED
+import com.braintreepayments.api.ShopperInsightsAnalytics.GET_RECOMMENDED_PAYMENTS_SUCCEEDED
+import com.braintreepayments.api.ShopperInsightsAnalytics.PAYPAL_PRESENTED
+import com.braintreepayments.api.ShopperInsightsAnalytics.PAYPAL_SELECTED
+import com.braintreepayments.api.ShopperInsightsAnalytics.VENMO_PRESENTED
+import com.braintreepayments.api.ShopperInsightsAnalytics.VENMO_SELECTED
+import java.lang.Exception
+
+/**
+ * Use [ShopperInsightsClient] to optimize your checkout experience
+ * by prioritizing the customer’s preferred payment methods in your UI.
+ * By customizing each customer’s checkout experience,
+ * you can improve conversion, increase sales/repeat buys and boost user retention/loyalty.
+ *
+ * Note: **This feature is in beta. It's public API may change in future releases.**
+ */
+class ShopperInsightsClient @VisibleForTesting internal constructor(
+    private val api: ShopperInsightsApi,
+    private val braintreeClient: BraintreeClient
+) {
+    constructor(braintreeClient: BraintreeClient) : this(
+        ShopperInsightsApi(EligiblePaymentsApi(braintreeClient)),
+        braintreeClient,
+    )
+
+    /**
+     * Retrieves recommended payment methods based on the provided shopper insights request.
+     *
+     * @param context Android context
+     * @param request The [ShopperInsightsRequest] containing information about the shopper.
+     * @return A [ShopperInsightsResult] object indicating the recommended payment methods.
+     * Note: This feature is in beta. Its public API may change or be removed in future releases
+     * PayPal recommendation is only available for US, AU, FR, DE, ITA, NED, ESP, Switzerland and
+     * UK merchants. Venmo recommendation is only available for US merchants.
+     */
+    fun getRecommendedPaymentMethods(
+        request: ShopperInsightsRequest,
+        callback: ShopperInsightsCallback
+    ) {
+        braintreeClient.sendAnalyticsEvent(GET_RECOMMENDED_PAYMENTS_STARTED)
+
+        if (request.email == null && request.phone == null) {
+            callbackFailure(
+                callback = callback,
+                error = IllegalArgumentException(
+                    "One of ShopperInsightsRequest.email or ShopperInsightsRequest.phone must be " +
+                            "non-null."
+                )
+            )
+            return
+        }
+
+        // TODO: get correct merchant ID from SDK
+        val merchantId = "MXSJ4F5BADVNS"
```

Comment:
Should this be updated based on the TODO?

Reply:
Good catch, yep it should be updated. Do you have a preference on if we update it as a part of this PR or a follow-up PR into main?

Reply:
Actually, i realized this should be removed. We're no longer sending the merchant ID as a param in the request body. The backend determines the merchant based on the auth fingerprint.

Reply:
> Good catch, yep it should be updated. Do you have a preference on if we update it as a part of this PR or a follow-up PR into main?

I think any changes should go in a new PR vs making the change directly in the feature branch PR before merging this branch into `main`. Generally our stance is that we do not require approvals on merging feature branches into `main` as all PRs that make up the feature branch should have been reviewed and received 2 approvals already. Opening the feature branch PR to `main` is mainly just a check that CI is all still green more than anything else.